Moonshot/baddies/Attributes.gd

```diff
@@ -0,0 +1,135 @@
+extends Node
+class_name Attributes
+
+onready var fade: Fade = $Fade
+
+enum Direction {
+	LEFT = -1,
+	RIGHT = 1
+}
+
+var dead = false
+var inital_hp: float
+var hp: float
+var gravity: int
+var speed: int = 230
+var direction: int = Direction.RIGHT
+var animation: String
+var velocity:= Vector2();
+var sprite: AnimatedSprite
+var floor_vector = Vector2(0, -1)
+var body: KinematicBody2D
+var damage_to_player: float = 0.2
+var should_damge_on_collision = false
+var shot_speed = 500
+var shot_damage = 0.5
+var item_drop = null
+var on_death: FuncRef
+
+func _ready():
+	fade.set_fade_speed(0.04)
+	fade.set_fade_factor(0.03)
+	fade.set_tree(get_tree())
+
+func get_current_state() -> Dictionary:
+	return {
+		"inital_hp": inital_hp,
+		"gravity": gravity,
+		"speed": speed,
+		"direction": direction,
+		"sprite": sprite,
+		"velocity": velocity,
+		"floor_vector": floor_vector,
+		"body": body,
+		"animation": animation,
+		"damage_to_player": damage_to_player,
+		"should_damge_on_collision": should_damge_on_collision,
+		"shot_speed": shot_speed,
+		"shot_damage": laser_damage,
+		"item_drop": item_drop,
```

Comment:
laser_damage needs to be defined in the class attributes to make it work 👍  :)

Reply:
Oh lol Bill already requested this change xD

Reply:
`"shot_damage": shot_damage`